client/src/components/PinnedContent/PinnedContent.tsx

```diff
@@ -16,25 +18,49 @@ import text from "./PinnedContent.yaml";
 
 const text_maker = create_text_maker(text);
 
-export const get_pinned_content_local_storage = (local_storage_name) => {
+export const get_pinned_content_local_storage = (
+  local_storage_name: string
+) => {
   try {
-    return has_local_storage && local_storage_name
-      ? JSON.parse(localStorage.getItem(local_storage_name))
-      : null;
+    if (has_local_storage) {
+      const storedString = localStorage.getItem(local_storage_name);
+      if (storedString !== null) {
+        return JSON.parse(storedString);
+      }
+    } else {
+      return null;
+    }
```

Comment:
Has a case with no explicit return now (which would be `undefined`, instead of the expected `null | boolean`). I'd suggest changing lines 30-32 to
```
}
return null;
```
So that the first conditional branch falls through to returning null if it doesn't otherwise return.